File: SUBS.py
for _ in range(int(input())):
	s1 = input()
	s2 = input()
	s1 = s1[:len(s1)-1]
	s2 = s2[:len(s2)-1]
	d1 = {}
	d2 = {}
	for i in s1:
		if i in d1.keys():
			d1[i] += 1
		else:
			d1[i] = 1
	for i in s2:
		if i in d2.keys():
			d2[i] += 1
		else:
			d2[i] = 1
	ans = 100077
	for i in d1.keys():
		c, val = i, d1[i]
		if c in d2.keys():
			ans = min(ans, d2[c] // val)
		else:
			ans = 0
	print(ans)

File: SUBS.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	int t;
	cin >> t;
	while (t--) {
		string s1, s2;
		cin >> s1 >> s2;
		map<char, int> m1, m2;
		for (int i = 0; i < s1.size(); i++)
			m1[s1[i]]++;
		for (int i = 0; i < s2.size(); i++)
			m2[s2[i]]++;
		int ans = INT_MAX;
		for (auto itr = m1.begin(); itr != m1.end(); itr++) {
			char c = itr->first;
			int val = itr->second;
			ans = min(ans, m2[c] / val);
		}
		cout << ans << endl;
	}
}
